test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_fill.cpp

```diff
@@ -0,0 +1,576 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "basic_command_buffer.h"
+#include "procs.h"
+#include <vector>
+
+//--------------------------------------------------------------------------
+template <bool check_image_support>
+struct CommandFillBaseTest : BasicCommandBufferTest
+{
+    CommandFillBaseTest(cl_device_id device, cl_context context,
+                        cl_command_queue queue)
+        : BasicCommandBufferTest(device, context, queue)
+    {}
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+        if (check_image_support)
+        {
+            src_image = create_image_2d(context, CL_MEM_WRITE_ONLY, &formats,
+                                        512, 512, 0, NULL, &error);
+            test_error(error, "create_image_2d failed");
+
+            dst_image = create_image_2d(context, CL_MEM_READ_ONLY, &formats,
+                                        512, 512, 0, NULL, &error);
+            test_error(error, "create_image_2d failed");
+        }
+
+        return CL_SUCCESS;
+    }
+
+    bool Skip() override
+    {
+        if (check_image_support)
+        {
+            cl_bool image_support;
+
+            cl_int error =
+                clGetDeviceInfo(device, CL_DEVICE_IMAGE_SUPPORT,
+                                sizeof(image_support), &image_support, nullptr);
+            test_error(error,
+                       "clGetDeviceInfo for CL_DEVICE_IMAGE_SUPPORT failed");
+
+            return (!image_support || BasicCommandBufferTest::Skip());
+        }
+        return BasicCommandBufferTest::Skip();
+    }
+
+protected:
+    clMemWrapper src_image;
+    clMemWrapper dst_image;
+    const cl_uint fill_color_1[4] = { 0x05, 0x05, 0x05, 0x05 };
+    const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+    const size_t origin[3] = { 0, 0, 0 };
+    const size_t region[3] = { 512, 512, 1 };
+    const cl_int pattern = 0xFF;
+};
+
+namespace {
+
+// CL_INVALID_COMMAND_QUEUE if command_queue is not NULL.
+struct CommandBufferCommandFillBufferQueueNotNull
+    : public CommandFillBaseTest<false>
+{
+    using CommandFillBaseTest::CommandFillBaseTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandFillBufferKHR(
+            command_buffer, queue, out_mem, &pattern, sizeof(cl_int), 0,
+            data_size(), 0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_COMMAND_QUEUE,
+                               "clCommandFillBufferKHR should return "
+                               "CL_INVALID_COMMAND_QUEUE",
+                               TEST_FAIL);
+
+        error = clCommandFillImageKHR(command_buffer, queue, src_image,
+                                      fill_color_1, origin, region, 0, nullptr,
+                                      nullptr, nullptr);
```

Comment:
We're using an image API here, but I think this is tested as part of `commandBufferCommandFillImageQueueNotNull`